src/lib.rs

```diff
@@ -139,14 +139,11 @@ pub async fn start_lemmy_server() -> Result<(), LemmyError> {
   );
 
   if scheduled_tasks_enabled {
+    let context = context.clone();
     // Schedules various cleanup tasks for the DB
-    thread::spawn({
-      let context = context.clone();
-      move || {
-        scheduled_tasks::setup(db_url, user_agent, context)
-          .expect("Couldn't set up scheduled_tasks");
-      }
-    });
+    scheduled_tasks::setup(db_url, user_agent, context)
+      .await
```

Comment:
nooooo you have to use something like `thread::spawn` (used in the removed code) or `tokio::task::spawn` so that `scheduled_tasks::setup` (which runs a loop forever) can run without blocking the execution of the rest of the `start_lemmy_server` function

Reply:
nice catch, I used `tokio::task::spawn`